### ISSUE TITLE
使用GPU训练的时候  利用率太低

### ISSUE THREAD

**Issue by qwe7771** [PaddlePaddle/PaddleClas#2260]
num_workers 设置16 但是还是单线程在载入数据  CPU使用率一致在25%

![image](https://user-images.githubusercontent.com/29864557/188530631-c57c976d-af9d-4b3a-ad87-f58a1eba177a.png)

# global configs
Global:
  checkpoints: 
  pretrained_model: "../pretrained/PPLCNet_x0_25_pretrained"
  output_dir: H:/jobs/2/3min
  device: gpu
  save_interval: 5
  eval_during_train: True
  eval_interval: 1
  epochs: 360
  print_batch_step: 10
  use_visualdl: False
  # used for static mode and model export
  image_shape: [3, 56, 56]
  save_inference_dir: H:/jobs/2/3min
# model architecture
Arch:
  name: PPLCNet_x0_25
  class_num: 5194
 
# loss function config for traing/eval process
Loss:
  Train:
    - CELoss:
        weight: 1.0
        epsilon: 0.1
  Eval:
    - CELoss:
        weight: 1.0

Optimizer:
  name: Momentum
  momentum: 0.9
  lr:
    name: Cosine
    learning_rate: 0.01
    warmup_epoch: 5
  regularizer:
    name: 'L2'
    coeff: 0.00003

# data loader for train and eval
DataLoader:
  Train:
    dataset:
      name: ImageNetDataset
      image_root: "H:/jobs/2/3min/samples"
      cls_label_path: "H:/jobs/2/3min/label-train.txt"
      transform_ops:
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - RandCropImage:
            size: 56
        - RandFlipImage:
            flip_code: 1
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
    sampler:
      name: DistributedBatchSampler
      batch_size: 640
      drop_last: False
      shuffle: True
    loader:
      num_workers: 16
      use_shared_memory: False

  Eval:
    dataset: 
      name: ImageNetDataset
      image_root: "H:/jobs/2/3min/samples"
      cls_label_path: "H:/jobs/2/3min/label-val.txt"
      transform_ops:
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - ResizeImage:
            resize_short: 56
        - CropImage:
            size: 56
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
    sampler:
      name: DistributedBatchSampler
      batch_size: 640
      drop_last: False
      shuffle: False
    loader:
      num_workers: 16
      use_shared_memory: False
Metric:
  Train:
    - TopkAcc:
        topk: [1, 5]
  Eval:
    - TopkAcc:
        topk: [1, 5]

**Comment by TingquanGao** [comment]
你好，可以进一步分析一下训练的log，其中有reader cost和batch cost两个字段，其中reader cost表示该batch的数据读取耗时，batch cost表示该batch的数据读取+前反向耗时。

**Comment by TingquanGao** [comment]
你好，可以进一步分析一下训练的log，其中有reader cost和batch cost两个字段，其中reader cost表示该batch的数据读取耗时，batch cost表示该batch的数据读取+前反向耗时。

**Comment by qwe7771** [comment]
> 你好，进一步分析一下，有reader cost和batch cost，reader costbatch的数据训练的两个批次的成本，可以表示该批次的数据读取+前批的其中一个批次，可以表示。

您看下这个 
[2022/09/06 09:56:48] ppcls INFO: [Train][Epoch 11/360][Iter: 280/548]lr: 0.00999, top1: 0.00231, top5: 0.01118, CELoss: 7.68446, loss: 7.68446, batch_cost: 0.43978s, reader_cost: 0.39589, ips: 1455.27340 images/sec, eta: 23:23:46
看这时间

**Comment by cuicheng01** [comment]
如果cpu核数不是很多，可以将num_workers调小一些，比如8或者4试试。另外训练这么小的模型可以尝试安装dali试试，安装完在Global.use_dali设置为True就好。

**Comment by qwe7771** [comment]
> 



> 如果 cpu 核不是很多，可以尝试将 num_worker 设置为真好。

这个 num_workers  无论怎么调整都没效果 还是1个线程

**Comment by cuicheng01** [comment]
> 这个 num_workers 无论怎么调整都没效果 还是1个线程

正常来说不太可能，你可以在后台看下进程数量。不过为了可以加速多一些，建议安装nvidia提供的dali包，这样这个模型大概可以加速四五倍。
